### ISSUE TITLE
TargetInvocationException:GoogleMobileAds.Android.RewardBasedVideoAdClient.onAdLoaded()

### ISSUE THREAD

**Issue by ChaleChan** [googleads/googleads-mobile-unity#824]
android mobile 
Sometimes, when ad loaded,
Rethrow as TargetInvocationException:GoogleMobileAds.Android.RewardBasedVideoAdClient.onAdLoaded()

stack:
GoogleMobileAds.Android.RewardBasedVideoAdClient:onAdLoaded()
TargetInvocationException:GoogleMobileAds.Android.RewardBasedVideoAdClient.onAdLoaded()
UnityEngine.AndroidJavaProxy.Invoke(System.String methodName,System.Object[] args)
UnityEngine.AndroidJavaProxy.Invoke(System.String methodName,UnityEngine.AndroidJavaObject[] javaArgs)
UnityEngine._AndroidJNIHelper.InvokeJavaProxyMethod(UnityEngine.AndroidJavaProxy proxy,IntPtr jmthodName,IntPtr jargs)

**Comment by rampara** [comment]
@ChaleChan can you provide a more complete exception log?

**Comment by ChaleChan** [comment]
> @ChaleChan can you provide a more complete exception log?

We have solved this problem already.
This problem might take place on Android mobile phones. After the request for AD, when the AD is loaded, the callback is not in the main thread. However, the Unity processing logic is in the main thread tho.
The temporary solution for this problem is that when the callback for AD occur, keep it cached, then call in main thread.

**Comment by rampara** [comment]
@ChaleChan you're correct in that ad events are not invoked on the main thread of Android. This [blog post](https://ads-developers.googleblog.com/2016/04/handling-android-ad-events-in-unity.html) provides more details.
